shared/js/ui/views/whitelist.es6.js

```diff
@@ -15,19 +15,84 @@ function Whitelist (ops) {
 Whitelist.prototype = window.$.extend({},
   Parent.prototype,
   {
+    _isHiddenClass: 'is-hidden',
+    _isDisabledClass: 'is-disabled',
+    _isInvalidInputClass: 'is-invalid-input',
 
     _removeItem: function (e) {
-      var itemIndex = window.$(e.target).data('item')
+      const itemIndex = window.$(e.target).data('item')
+      const $listItem = window.$(e.target).parent()
       this.model.removeDomain(itemIndex)
-      this.setWhitelistFromSettings()
+
+      // No need to rerender the whole view
+      $listItem.remove()
```

Comment:
We used to rerender the whole view here. 
Not ok anymore because if the input is visible, by rerendering it we lose that state (and the input's content), which is a confusing behavior.